doc/api/functions.rst

```diff
@@ -0,0 +1,11 @@
+API Documentation: Functions
```

Comment:
I think "Functions" isn't really a meaningful category.

Probably "Convenience Functions using NumPy Arrays" would be a relevant category?

Reply:
I changed to "Top-level functions".

I think that "convenience functions" is terminology that puts the point of view of the developer higher than the point of view of the user. As the developer, you see the main part of `sounddevice` as the complex stream class hierarchy and the integration with portaudio. But for the average user, `play` and `rec` are going to be the most important things, and all the rest are things under the hood that you can use if you want to really customize how things work.

That's my opinion, if you feel otherwise I'll change to whichever title you'd pick.

Reply:
There have been multiple questions (e.g. on stackoverflow) in the past, where users were using `play()` et al. for situation where they shouldn't be used. So no, I don't agree, this separation/grouping is very much relevant for "users". That's also the reason why I recently did e8253a615d6f70f7a1f6bc662e703a857fd2e8c4, where I tried to point out the limitations of those functions.

I *do* agree that `play()` is the most important function for most people, but I think it is currently sufficiently visible.

Anyway, we can postpone the naming of the page for a moment.

More importantly, the "convenience functions" should be on a page of their own, not intermixed with the "low level" wrapper functions.

Reply:
Okay, I changed to the name you suggested. I broke it down into two pages, I'm not sure I got the exact division you meant, feel free to point out anything you want me to move.